botocore/loaders.py

```diff
@@ -221,6 +242,93 @@ def create_loader(search_path_string=None):
     return Loader(extra_search_paths=paths)
 
 
+def _create_path(path):
+    """Construct a BotoZipPath or pathlib.Path object from a path string.
+
+    Split the root (path to a zip) and the path to the subdirectory/file e.g.
+    '/Users/user123/Documents/Archive.zip/data' -->
+    BotoZipPath('/Users/user123/Documents/Archive.zip', '/data')
+    If the zipped path doesn't exist, return a pathlib.Path object instead
+
+    :type path: str
+    :param path: Full path to construct a BotoZipPath or pathlib.Path object
+
+    :return: A BotoZipPath/pathlib.Path object consisting of the root and
+    the path to a file or subdirectory
+
+    """
+    path_obj = pathlib.Path(path)
+    resolved_path = path_obj.resolve()
+    parts = resolved_path.parts
+    new_path = ''
+    while len(parts) > 0:
```

Comment:
This lets us short circuit the whole zip construction flow and solves the Python 3.7/3.8 issue I asked about earlier.

```suggestion
    while parts and HAS_ZIP_SUPPORT:
```

Reply:
I'm going to keep `while len(parts) > 0` unless you had a specific reason to remove?